src/drivers/Packet.py

```diff
@@ -0,0 +1,68 @@
+# Written by Walden Hillegass 
+# details of packet structure are in this document :
+# https://docs.google.com/document/d/e/2PACX-1vRbm1WKCdh5X4Ji4nGYI22LL2Y-oLjkN6hBZEl6U7xsOcApdzz1ynh-aBq82YZGA1eBMbsIvnchxTQM/pub
+
+import hashlib
+MAX_PAYLOAD = 500
+OVERHEAD = 12
+
+class PacketizerException(Exception):
+    pass
+
+class Packet:
+    def __init__(self, packet_id : int, payload : bytes):
+        if (len(payload) > MAX_PAYLOAD):
+            raise PacketizerException("Payload is larger than max size. Use multi_packetizer() instead")
+        id_bytes = packet_id.to_bytes(4, 'little')
+        payload_length = len(payload).to_bytes(2, 'little')
+        b = []
+        b.append(b'\x69\x69')
+        b.append(id_bytes)
+        b.append(payload_length)
+        b.append(hashlib.sha256(id_bytes + payload).digest()[:4])
+        b.append(payload)
+
+        self.bytes = b"".join(b)
+        
+
+# Returns a list of packets
+def multi_packetizer(payload : bytes, starting_packet_id : int) -> list:
+    packets = []
+    packet_id = starting_packet_id
+    while len(payload) > 0:
+        if len(payload) > MAX_PAYLOAD:
+            packets.append(Packet(packet_id, payload[:MAX_PAYLOAD]))
+            payload = payload[MAX_PAYLOAD:]
+        else:
+            packets.append(Packet(packet_id, payload))
+            payload = b""
+        packet_id += 1
+    return packets
```

Comment:
Could we combine these two into one? Where we always return a list of packets, and the method determines whether we need to split it across multiple packets?

It would just make the abstraction a bit cleaner to use.